anndata/_core/sparse_dataset.py

```diff
@@ -343,7 +343,29 @@ def __repr__(self) -> str:
     def __getitem__(self, index: Index | tuple[()]) -> float | ss.spmatrix:
         row, col = self._normalize_index(index)
         mtx = self._to_backed()
-        sub = mtx[row, col]
+
+        # see https://github.com/scverse/anndata/issues/1224 for special handling of boolean masks
+        # scipy internally converts boolean masks to integer indices so this needs to occur before access to `mtx`.
```

Comment:
While this can't go into one of the methods in the `BackedSparseMatrix` classes, could this logic at least be factored out into a function? Mainly for readability of this method, and so it's easy to move later.

Reply:
Tied, see above comment about refactoring.